modules/node-shared/src/main/scala/io/constellationnetwork/node/shared/domain/delegatedStake/UpdateDelegatedStakeValidator.scala

```diff
@@ -0,0 +1,198 @@
+package io.constellationnetwork.node.shared.domain.delegatedStake
+
+import cats.data.{NonEmptySet, ValidatedNec}
+import cats.effect.Async
+import cats.syntax.all._
+
+import scala.collection.immutable.{SortedMap, SortedSet}
+
+import io.constellationnetwork.ext.cats.syntax.validated._
+import io.constellationnetwork.node.shared.domain.delegatedStake.UpdateDelegatedStakeValidator.UpdateDelegatedStakeValidationErrorOr
+import io.constellationnetwork.node.shared.domain.seedlist.SeedlistEntry
+import io.constellationnetwork.schema.address.Address
+import io.constellationnetwork.schema.delegatedStake.{CreateDelegatedStakeReference, DelegatedStakeAmount, UpdateDelegatedStake}
+import io.constellationnetwork.schema.epoch.EpochProgress
+import io.constellationnetwork.schema.nodeCollateral.UpdateNodeCollateral
+import io.constellationnetwork.schema.peer.PeerId
+import io.constellationnetwork.schema.tokenLock.{TokenLock, TokenLockReference}
+import io.constellationnetwork.schema.{GlobalSnapshotInfo, SnapshotOrdinal, nodeCollateral}
+import io.constellationnetwork.security.signature.SignedValidator.SignedValidationError
+import io.constellationnetwork.security.signature.signature.SignatureProof
+import io.constellationnetwork.security.signature.{Signed, SignedValidator}
+import io.constellationnetwork.security.{Hasher, SecurityProvider}
+
+import derevo.cats.{eqv, show}
+import derevo.derive
+import eu.timepit.refined.types.numeric.NonNegLong
+
+trait UpdateDelegatedStakeValidator[F[_]] {
+  def validateCreateDelegatedStake(
+    signed: Signed[UpdateDelegatedStake.Create],
+    lastContext: GlobalSnapshotInfo,
+    lastGlobalEpochProgress: EpochProgress
+  ): F[UpdateDelegatedStakeValidationErrorOr[Signed[UpdateDelegatedStake.Create]]]
+  def validateWithdrawDelegatedStake(
+    signed: Signed[UpdateDelegatedStake.Withdraw],
+    lastContext: GlobalSnapshotInfo
+  ): F[UpdateDelegatedStakeValidationErrorOr[Signed[UpdateDelegatedStake.Withdraw]]]
+}
+
+object UpdateDelegatedStakeValidator {
+  def make[F[_]: Async: SecurityProvider](
+    signedValidator: SignedValidator[F],
+    seedlist: Option[Set[SeedlistEntry]],
+    minEpochProgressesToLock: NonNegLong
+  )(implicit hasher: Hasher[F]): UpdateDelegatedStakeValidator[F] =
+    new UpdateDelegatedStakeValidator[F] {
+
+      def validateCreateDelegatedStake(
+        signed: Signed[UpdateDelegatedStake.Create],
+        lastContext: GlobalSnapshotInfo,
+        lastGlobalEpochProgress: EpochProgress
+      ): F[UpdateDelegatedStakeValidationErrorOr[Signed[UpdateDelegatedStake.Create]]] =
+        for {
+          numberOfSignaturesV <- validateNumberOfSignatures(signed)
```

Comment:
Nitpick: not required to change, but most of the functionality in these validators is shared, any way we can abstract here?